plugins/fetcher/prometheus/fetcher.go

```diff
@@ -0,0 +1,146 @@
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package prometheus
+
+import (
+	"context"
+	"time"
+
+	"github.com/apache/skywalking-satellite/internal/pkg/config"
+	"github.com/apache/skywalking-satellite/internal/pkg/log"
+	"github.com/apache/skywalking-satellite/internal/satellite/event"
+
+	promConfig "github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/scrape"
+	yaml "gopkg.in/yaml.v3"
+	v1 "skywalking.apache.org/repo/goapi/satellite/data/v1"
+)
+
+const (
+	Name      = "prometheus-metrics-fetcher"
+	eventName = "prometheus-metrics-event"
+)
+
+type scrapeConfig struct {
+	JobName        string                   `yaml:"job_name" mapstructure:"job_name"`
+	ScrapeInterval time.Duration            `yaml:"scrape_interval,omitempty" mapstructure:"scrape_interval,omitempty"`
+	StaticConfigs  []map[string]interface{} `yaml:"static_configs" mapstructure:"static_configs"`
+	MetricsPath    string                   `yaml:"metrics_path,omitempty" mapstructure:"metrics_path,omitempty"`
+}
+
+// Fetcher is the struct for Prometheus fetcher
+type Fetcher struct {
+	config.CommonFields
+	// config is the top level configuratScrapeConfigsMapion of prometheus
+	ScrapeConfigsMap []*scrapeConfig `mapstructure:"scrape_configs" yaml:"scrape_configs"`
+
+	ScrapeConfigs []*promConfig.ScrapeConfig
+	// events
+	OutputEvents event.BatchEvents
+	// outputChannel
+	OutputChannel chan *v1.SniffData
+
+	cancelFunc context.CancelFunc
+}
+
+func (f *Fetcher) Name() string {
+	return Name
+}
+
+func (f *Fetcher) Description() string {
+	return "This is a fetcher for Skywalking prometheus metrics format, " +
+		"which will translate Prometheus metrics to Skywalking meter system."
+}
+
+func (f *Fetcher) DefaultConfig() string {
+	return `
+## some config here
+scrape_configs:
+ - job_name: 'prometheus'
+   metrics_path: '/metrics'
+   scrape_interval: 10s
+   static_configs:
+   - targets: ['127.0.0.1:2020']
+`
+}
+
+func (f *Fetcher) Prepare() {}
+
+func (f *Fetcher) Fetch() event.BatchEvents {
+	ctx, cancel := context.WithCancel(context.Background())
+	f.cancelFunc = cancel
+	// yaml
+	configDeclare := make(map[string]interface{})
+	configDeclare["scrape_configs"] = f.ScrapeConfigsMap
+	configBytes, err := yaml.Marshal(configDeclare)
+	if err != nil {
+		log.Logger.Fatal("prometheus fetcher configure failed", err.Error())
```

Comment:
should we return here? since it parse failed.

Reply:
It is `Fatal` , which will exit the application.